tests/integration_tests.rs

```diff
@@ -0,0 +1,844 @@
+                "list_ref": list_id,
+                "max_payments": 100
+            }),
+        )
+        .unwrap()
+        .transaction()
+        .gas(near_sdk::Gas::from_tgas(300))
+        .with_signer(user_id.clone(), user_signer.clone())
+        .send_to(&network_config)
+        .await
+        .unwrap()
+        .assert_success();
+
+    // Process third batch (50 payments)
+    near_api::Contract(contract_id.clone())
+        .call_function(
+            "payout_batch",
+            json!({
+                "list_ref": list_id,
+                "max_payments": 100
+            }),
+        )
+        .unwrap()
+        .transaction()
+        .gas(near_sdk::Gas::from_tgas(300))
+        .with_signer(user_id.clone(), user_signer.clone())
+        .send_to(&network_config)
+        .await
+        .unwrap()
+        .assert_success();
+
+    // Verify all recipients received their payments
+    for (i, recipient) in recipients.iter().enumerate() {
+        let balance = near_api::Account(recipient.clone())
+            .view()
+            .fetch_from(&network_config)
+            .await
+            .unwrap()
+            .data
+            .amount;
+        
+        // Each recipient started with 1 NEAR and should have received 1 NEAR
+        // So balance should be approximately 2 NEAR (minus some gas fees)
+        assert!(
+            balance.as_yoctonear() > 1_500_000_000_000_000_000_000_000, // At least 1.5 NEAR
+            "Recipient {} should have received payment, balance: {} yoctoNEAR",
+            i,
+            balance.as_yoctonear()
+        );
+    }
+
+    // Verify contract balance is not less than before (storage revenue retained)
+    let contract_balance_after = near_api::Account(contract_id.clone())
+        .view()
+        .fetch_from(&network_config)
+        .await
+        .unwrap()
+        .data
+        .amount;
+
+    assert!(
+        contract_balance_after.as_yoctonear() >= contract_balance_before.as_yoctonear(),
+        "Contract balance should not decrease after payouts. Before: {}, After: {}",
+        contract_balance_before.as_yoctonear(),
+        contract_balance_after.as_yoctonear()
+    );
+
+    // Verify all payments are marked as Paid
+    let list: serde_json::Value = near_api::Contract(contract_id.clone())
+        .call_function("view_list", json!({ "list_ref": list_id }))
+        .unwrap()
+        .read_only()
+        .fetch_from(&network_config)
+        .await
+        .unwrap()
+        .data;
+
+    let payments_array = list["payments"].as_array().unwrap();
+    assert_eq!(payments_array.len(), 250, "Should have 250 payments");
+    
+    for (i, payment) in payments_array.iter().enumerate() {
+        assert_eq!(
+            payment["status"], "Paid",
+            "Payment {} should be marked as Paid",
+            i
+        );
+    }
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_failed_payment_retry() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
Can you rather make this failed payment retry test to be about recipient accounts that does not exist. Then you can create the account before retrying and see that it works.